src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -114,6 +130,72 @@ public Localizable getActionBlurb() {
         return SupportPlugin.getComponents();
     }
 
+    public class Bundle {
+        File location;
+        public Bundle(File location) {
+            this.location = location;
+        }
+
+        public File getFile() {
+            return location;
+        }
+
+        public String getName() {
+            return location.getName();
+        }
+
+        public String getHref() {
+            return "bundles/bundle/" + getName();
+        }
+
+        public long getLength() {
+            return location.length();
+        }
+
+        private void doDownload(StaplerRequest req, StaplerResponse rsp) throws IOException {
+            OutputStream os = rsp.getOutputStream();
+            try {
+                rsp.setContentType("application/zip");
+                Util.copyStreamAndClose(new FileInputStream(location), os);
+            } finally {
+                IOUtils.closeQuietly(os);
+            }
+        }
+    }
+
+    public final class BundleList extends ArrayList<Bundle> {
```

Comment:
Should be static as does not use any out class methods.

Also not convinced that this should extend the Array List, rather than have a method in the parent and just store a `ArrayList<Bundle>` in the parent